common/thrift_router.h

```diff
@@ -377,9 +383,20 @@ class ThriftRouter {
         const Role role,
         const unsigned rotation_counter,
         const std::string& segment,
-        const int shrink_target) {
+        const int shrink_target,
+        const std::string& specific_az) {
```

Comment:
sorry, just realize one more thing: if user specified a AZ, and in the meantime wants a role `master/leader`, the currently implementation will ignore the role completely, but that may be surprising. Should we take `role` into consideration when doing AZ aware routing?

Reply:
yes, I added the code which will take role into consideration

Reply:
The comments of this function says we firstly filter hosts by role, then sort them by other preferences, which I think makes sense. However, the implementation of the AZ aware routing firstly filters by AZ, then sort them by role

Reply:
fixed it

Reply:
instead of adding separate condition, I am checking of specific_az right before inserting into the host array.